lldb/source/Core/TelemetryVendor.cpp

```diff
@@ -0,0 +1,46 @@
+//===-- TelemetryVendor.cpp -----------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+#ifdef LLDB_BUILD_TELEMETRY
+
+#include "lldb/Core/TelemetryVendor.h"
+
+namespace lldb_private {
+
+llvm::StringRef TelemetryVendor::GetPluginName() {
+  return "UpstreamTelemetryVendor";
+}
+
+void TelemetryVendor::Initialize() {
+  // The default (upstream) impl will have telemetry disabled by default.
+  SetTelemetryConfig(std::make_shared<new llvm::telemetry::Config>(
+      /*enable_telemetry*/ false));
+  SetTelemetryManager(nullptr);
+}
```

Comment:
Could we make it so that we don't need this method? So that the values of these two objects are initialized to this by default?

Reply:
done

Reply:
That's better, but can we now remove the method altogether (instead of just leaving it empty)? There should be no need to "initialize" the base class if it's not doing anything. Ideally, only the concrete vendor subclasses should need initialization (that is also how (most) our other plugins work, though it may not be the case with SymbolVendor, which I suspect you're drawing inspiration from))

Reply:
Done. removed this part.